jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricher.java

```diff
@@ -157,58 +151,111 @@ private boolean hasExactlyOneServicePort(ServiceBuilder service, String id) {
 
         if (serviceMetadata != null && StringUtils.isNotBlank(serviceMetadata.getName())
                 && hasExactlyOneServicePort(serviceBuilder, serviceMetadata.getName()) && isExposedService(serviceMetadata)) {
             String name = serviceMetadata.getName();
-            if (!hasRoute(listBuilder, name)) {
-                if (StringUtils.isNotBlank(routeDomainPostfix)) {
-                    routeDomainPostfix = prepareHostForRoute(routeDomainPostfix, name);
-                } else {
-                    routeDomainPostfix = "";
+            updateRouteDomainPostFixBasedOnServiceName(name);
+            Route opinionatedRoute = createOpinionatedRouteFromService(serviceBuilder, routeDomainPostfix);
+            if (opinionatedRoute != null) {
+                int routeFromFragmentIndex = getRouteIndexWithName(listBuilder, name);
+                if (routeFromFragmentIndex > 0) { // Merge fragment with Opinionated Route
+                    Route routeFragment = (Route) listBuilder.buildItems().get(routeFromFragmentIndex);
+                    Route mergedRoute = mergeRoute(routeFragment, opinionatedRoute);
+                    KubernetesResourceUtil.removeItemFromKubernetesBuilder(listBuilder, listBuilder.getItems().get(routeFromFragmentIndex));
+                    listBuilder.addToItems(mergedRoute);
+                } else { // No fragment provided. Use Opinionated Route.
+                    listBuilder.addToItems(opinionatedRoute);
                 }
+            }
+        }
+    }
 
-                RoutePort routePort = createRoutePort(serviceBuilder);
-                if (routePort != null) {
-                    RouteBuilder routeBuilder = new RouteBuilder().
-                            withMetadata(serviceMetadata).
-                            withNewSpec().
-                            withPort(routePort).
-                            withNewTo().withKind("Service").withName(name).endTo().
-                            withHost(routeDomainPostfix.isEmpty() ? null : routeDomainPostfix).
-                            endSpec();
-
-                    // removing `expose : true` label from metadata.
-                    removeLabel(routeBuilder.buildMetadata(), EXPOSE_LABEL, "true");
-                    removeLabel(routeBuilder.buildMetadata(), JKubeAnnotations.SERVICE_EXPOSE_URL.value(), "true");
-                    routeBuilder.withNewMetadataLike(routeBuilder.buildMetadata());
-                    routes.add(routeBuilder.build());
-                }
+    private void updateRouteDomainPostFixBasedOnServiceName(String serviceName) {
+        if (StringUtils.isNotBlank(routeDomainPostfix)) {
+            routeDomainPostfix = prepareHostForRoute(routeDomainPostfix, serviceName);
+        } else {
+            routeDomainPostfix = "";
+        }
+    }
+
+    static Route mergeRoute(Route routeFromFragment, Route opinionatedRoute) {
+        // Update ApiVersion to route.openshift.io/v1
```

Comment:
This comment is completely irrelevant.
If you want to add a helpful comment because it's necessary and cannot be inferred with a test, then write something like
`// Update version because of/due to....`

I read the code and still don't understand why the change in the api version is required. This comment only tells me what I can already read in the code.

Reply:
https://github.com/eclipse/jkube/issues/383